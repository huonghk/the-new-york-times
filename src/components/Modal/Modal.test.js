import React from 'react';
import { shallow } from 'enzyme';
import { Modal, ModalTrigger, ModalContent } from './index';

describe('Modal Component', () => {

  const wrapper = shallow(
    <Modal>
      <ModalTrigger>Open Modal</ModalTrigger>
      <ModalContent>Modal Content</ModalContent>
    </Modal>
  );
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
