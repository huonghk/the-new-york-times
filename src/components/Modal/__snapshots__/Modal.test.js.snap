// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    shouldCloseOnClickOutside={true}
    shouldCloseOnPressEscape={true}
  >
    <ModalTrigger>
      Open Modal
    </ModalTrigger>
    <ModalContent
      role="dialog"
    >
      Modal Content
    </ModalContent>
  </Modal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <button
          className="trigger"
          onClick={[Function]}
        >
          Open Modal
        </button>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Open Modal",
          "className": "trigger",
          "onClick": [Function],
        },
        "ref": [Function],
        "rendered": "Open Modal",
        "type": "button",
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            className="trigger"
            onClick={[Function]}
          >
            Open Modal
          </button>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Open Modal",
            "className": "trigger",
            "onClick": [Function],
          },
          "ref": [Function],
          "rendered": "Open Modal",
          "type": "button",
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
